### PR TITLE
chore(sdk): Many clean ups in `SlidingSync`

### PR DESCRIPTION
**Pull request by Hywan**
I suggest to read this patch commit by commit, as it's a set of small local changes.

### CODE REVIEW

File: crates/matrix-sdk/src/sliding_sync/builder.rs [hunk 1]
```diff
@@ -223,7 +226,8 @@ impl SlidingSyncBuilder {
 
     /// Build the Sliding Sync.
     ///
-    /// If configured, load the cached data from cold storage.
+    /// If `self.storage_key` has some value, load the cached data from
```
Comment (jplatte) [inline crates/matrix-sdk/src/sliding_sync/builder.rs:229]: "has some value" => "is `Some(_)`"

File: crates/matrix-sdk/src/sliding_sync/room.rs [hunk 2]
```diff
@@ -325,6 +312,22 @@ impl From<&SlidingSyncRoom> for FrozenSlidingSyncRoom {
     }
 }
 
+impl From<(FrozenSlidingSyncRoom, Client)> for SlidingSyncRoom {
```
Comment (jplatte) [inline crates/matrix-sdk/src/sliding_sync/room.rs:307]: I'm not a fan of this change. `From` impls on tuples are a bit magic. Also this makes the conversion part of the public interface, it was private before.

File: crates/matrix-sdk/src/sliding_sync/room.rs [hunk 3]
```diff
@@ -77,16 +78,18 @@ impl SlidingSyncRoom {
 
     fn timeline_builder(&self) -> Option<TimelineBuilder> {
         if let Some(room) = self.client.get_room(&self.room_id) {
-            let timeline_queue = self.timeline_queue.lock_ref().to_vec();
-            let prev_batch = self.prev_batch.lock_ref().clone();
-            Some(Timeline::builder(&room).events(prev_batch, timeline_queue))
+            Some(
+                Timeline::builder(&room)
+                    .events(self.prev_batch.get_cloned(), self.timeline_queue.lock_ref().to_vec()),
```
Comment (jplatte) [inline crates/matrix-sdk/src/sliding_sync/room.rs:83]: `get_cloned` also locks, internally. But doesn't hurt 🤷🏼

File: crates/matrix-sdk/src/sliding_sync/room.rs [hunk 4]
```diff
@@ -150,23 +150,12 @@ impl SlidingSyncRoom {
...
-        }
-        if !required_state.is_empty() {
-            self.inner.required_state = required_state.clone();
-        }
+        self.inner.unread_notifications = unread_notifications;
+        self.inner.name = name;
+        self.inner.initial = initial;
+        self.inner.is_dm = is_dm;
+        self.inner.invite_state = invite_state;
+        self.inner.required_state = required_state;
```
Comment (jplatte) [inline crates/matrix-sdk/src/sliding_sync/room.rs:158]: ❤️

**Review by jplatte — CHANGES_REQUESTED** [review]

File: crates/matrix-sdk/src/sliding_sync/room.rs (hunk 3 above)
Comment (Hywan) [reply crates/matrix-sdk/src/sliding_sync/room.rs:83]: Ah OK, but yeah, it doesn't hurt.

File: crates/matrix-sdk/src/sliding_sync/room.rs (hunk 4 above)
Comment (Hywan) [reply crates/matrix-sdk/src/sliding_sync/room.rs:158]: 🙇

File: crates/matrix-sdk/src/sliding_sync/builder.rs (hunk 1 above)
Comment (Hywan) [reply crates/matrix-sdk/src/sliding_sync/builder.rs:229]: :+1:

File: crates/matrix-sdk/src/sliding_sync/room.rs (hunk 2 above)
Comment (Hywan) [reply crates/matrix-sdk/src/sliding_sync/room.rs:307]: OK, let's rollback to what we had previously.

**Review by jplatte — APPROVED** [review]
I would prefer if the reverted commit was force-pushed away, but changes look good now anyways.

**Pull request merged**